kong/db/strategies/postgres/plugins.lua

```diff
@@ -0,0 +1,57 @@
+local split = require("pl.stringx").split
+
+
+local insert = table.insert
+
+
+local Plugins = {}
+
+
+-- Emulate the `select_by_cache_key` operation
+-- using the `plugins` table of a 0.14 database.
+-- @tparam string key a 0.15+ plugin cache_key
+-- @treturn table|nil,err the row for this unique cache_key
+-- or nil and an error object.
+function Plugins:select_by_cache_key_migrating(key)
+  -- unpack cache_key
+  local parts = split(key, ":")
+
+  -- build query and args
+  local qbuild = { "SELECT * FROM plugins WHERE name = '" .. parts[2] .. "'" }
+  for i, field in ipairs({
+    "route_id",
+    "service_id",
+    "consumer_id",
+    "api_id",
+  }) do
+    local id = parts[i + 2]
+    if id ~= "" then
+      insert(qbuild, field .. " = '" .. id .. "'")
+    else
+      insert(qbuild, field .. " IS NULL")
+    end
+  end
+  local query = table.concat(qbuild, " AND ")
+
+  -- perform query, trying both temp and old table
+  local errs = 0
+  local last_err
+  for row, err in self.connector:iterate(query) do
```

Comment:
Do we need an iterator at all in this strategy? Since this is using SQL's `AND` condition instead of CQL's, are we ever going to get more than one row as a result to this query? The below condition (`if row then`) seems to indicate that it isn't the case.